packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -430,9 +444,30 @@ export class FluidDataStoreRuntime
 		this.identifyLocalChangeInSummarizer("DDSCreatedInSummarizer", id, type);
 	}
 
-	public createChannel(id: string = uuid(), type: string): IChannel {
-		if (id.includes("/")) {
-			throw new UsageError(`Id cannot contain slashes: ${id}`);
+	public createChannel(idArg: string | undefined, type: string): IChannel {
+		let id: string;
+
+		if (idArg !== undefined) {
+			id = idArg;
+			this.validateChannelId(id);
+		} else {
+			// We use three non-overlapping namespaces:
+			// - detached state: even numbers
+			// - attached state: odd numbers
+			// - uuids
+			// In first two cases we will encode result as strings in more compact form, with leading underscore,
+			// to ensure no overlap with user-provided DDS names (see validateChannelId())
+			if (this.visibilityState !== VisibilityState.GloballyVisible) {
```

Comment:
Should't this be `this.attachState` to maintain parity with the logic at `DataStores`?

Reply:
To be honest, I got lost in all these states.
I do not have VisibilityState in DataStores - I think this is something @andre4i will need to figure out how to add (to control same states for nested data stores). At the moment, data stores only deal with container-wide states.
 
If I got it right,
VisibilityState.GloballyVisible maps to Attached & Attaching (container is attached or attaching).
VisibilityState.LocallyVisible -> AttachState.Detached (attached to container, but container is not attached)
VisibilityState.NotVisible - not attached to container (is it still same AttachState.Detached)?

Does this look right?
If yes, can switch to using AttachState.Detached, but I'm not sure if it buys us anything.
I actually like more VisibilityState wording, especially given that almost no code should care about the difference between attaching & attached.

Reply:
@agarwal-navin - I'm a bit afraid to touch it, so I'd love to keep it as is. I'd love all layers to move to visibility language - I think it's way clearer.

Reply:
Yes, this is fine. VisibilityState is clearer to me as well. Let's keep that.